app/models/user.rb

```diff
@@ -9,41 +9,41 @@ class User < ApplicationRecord
   has_many :posts
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
-  has_many :received_invitations, class_name: 'FriendshipInvitation', foreign_key: 'invitee_id'
-  has_many :sent_invitations, class_name: 'FriendshipInvitation', foreign_key: 'inviter_id'
+  has_many :inverted_friendships, class_name: 'Friendship', foreign_key: 'friend_id', dependent: :destroy
+  has_many :friendships, class_name: 'Friendship', foreign_key: 'user_id', dependent: :destroy
 
   def friends
-    friends = received_invitations.map { |invitation| invitation.inviter if invitation.confirmed == true }
-    friends += sent_invitations.map { |invitation| invitation.invitee if invitation.confirmed == true }
-    friends.compact
+    friendships.map { |invitation| invitation.friend if invitation.confirmed == true }.compact
   end
 
-  def pending_invitations
-    sent_invitations.map { |invitation| invitation.invitee if invitation.confirmed == false }.compact
+  def pending_friends
+    friendships.map { |invitation| invitation.friend if invitation.confirmed == false }.compact
   end
 
   def friendship_requests
-    received_invitations.map { |invitation| invitation.inviter if invitation.confirmed == false }.compact
+    inverted_friendships.map { |invitation| invitation.user if invitation.confirmed == false }.compact
   end
```

Comment:
- [ ] you don't need to use a function to get friendship_requests since it can be done simply by the relationship between the user table and the friendship table.
```ruby
# Users who requested to be friends (needed for notifications)
has_many :inverted_friendships, -> { where confirmed: false }, class_name: "Friendship", foreign_key: "friend_id"
has_many :friend_requests, through: :inverted_friendships
```